dGame/dComponents/PetComponent.cpp

```diff
@@ -794,23 +657,335 @@ void PetComponent::Wander() {
+	Entity* const owner = GetOwner();
+	if (!owner) return;
+
+	constexpr auto petAbility = ePetAbilityType::Invalid;
+
+	m_Interaction.obj = LWOOBJID_EMPTY;
+	m_Interaction.type = PetInteractType::none;
+	m_Interaction.ability = petAbility;
+	SetPetAiState(PetAiState::follow);
+	m_Flags.Reset<PetFlag::IDLE>();
+	SetIsReadyToInteract(false);
+	SetIsHandlingInteraction(false); // Needed?
+	m_MovementAI->SetMaxSpeed(m_PetInfo.sprintSpeed);
+	m_MovementAI->SetHaltDistance(m_FollowRadius);
+	LOG_DEBUG("Stopping interaction!");
+
+	if (!bDontSerialize) {
+		Game::entityManager->SerializeEntity(m_Parent);
+	}
+
+	GameMessages::SendShowPetActionButton(m_Owner, petAbility, false, owner->GetSystemAddress()); // Needed?
+}
+
+void PetComponent::SetupInteractBouncer() {
+	const auto* const owner = GetOwner();
+	if (!owner) return;
+
+	LOG_DEBUG("Setting up bouncer interaction!");
+	SetIsReadyToInteract(true);
+	constexpr auto petAbility = ePetAbilityType::JumpOnObject;
+
+	m_Interaction.ability = petAbility;
+	m_Flags.Unset<PetFlag::IDLE>();
+	m_Flags.Set<PetFlag::ON_SWITCH, PetFlag::NOT_WAITING>(); // TODO: Double-check this is the right flag being set
+	LOG_DEBUG("m_Flags = %d", m_Flags);
+	Game::entityManager->SerializeEntity(m_Parent); // TODO: Double-check pet packet captures
+
+	const auto sysAddr = owner->GetSystemAddress();
+	GameMessages::SendHelp(m_Owner, eHelpType::PR_BOUNCER_TUTORIAL_03, sysAddr);
+	GameMessages::SendShowPetActionButton(m_Owner, petAbility, true, sysAddr);
+
+	SwitchComponent* closestSwitch = SwitchComponent::GetClosestSwitch(m_MovementAI->GetDestination()); // TODO: Find a better way to do this
+	closestSwitch->EntityEnter(m_Parent);
+
+	m_Timer += 0.5f;
+}
+
+void PetComponent::StartInteractBouncer() {
+	Entity* const user = GetOwner();
+	if (IsHandlingInteraction() || !user) return;
+
+	auto* const destroyableComponent = user->GetComponent<DestroyableComponent>();
+	if (!destroyableComponent) return;
+
+	auto imagination = destroyableComponent->GetImagination();
+	const auto imaginationCost =
+		CDClientManager::GetTable<CDPetAbilitiesTable>()->GetByID(ePetAbilityType::JumpOnObject).imaginationCost;
+
+	if (imagination < imaginationCost) {
+		//GameMessages::SendHelp(user->GetObjectID(), eHelpType::PR_NEED_IMAGINATION, user->GetSystemAddress()); // Check if right message!
+		return;
+	}
+	GameMessages::SendHelp(user->GetObjectID(), eHelpType::PR_TOOLTIP_1ST_PET_JUMPED_ON_SWITCH, user->GetSystemAddress());
+
+	GameMessages::SendShowPetActionButton(m_Owner, ePetAbilityType::Invalid, false, user->GetSystemAddress());
+
+	imagination -= imaginationCost;
+	destroyableComponent->SetImagination(imagination);
+	Game::entityManager->SerializeEntity(user);
+
+	// THIS IS ALL BAD, BAD, BAD! FIX IT, ME! >:(
+	SetIsHandlingInteraction(true);
+	SwitchComponent* closestSwitch = SwitchComponent::GetClosestSwitch(m_MovementAI->GetDestination()); // TODO: Find a better way to do this
+	closestSwitch->EntityEnter(m_Parent);
```

Comment:
This should be calling OnUse to make sure faction filtering is done

Reply:
`// THIS IS ALL BAD, BAD, BAD! FIX IT, ME! >:(` lol, I forgot that was there
Anyway, I'll make sure to change the call 👍